pages/profile/[username].tsx

```diff
@@ -1,35 +1,37 @@
 import * as React from 'react'
+import { useQuery } from '@apollo/react-hooks'
 import _ from 'lodash'
 import Layout from '../../components/Layout'
+import { useRouter } from 'next/router'
 import { Lesson } from '../../@types/lesson'
-import { AppData } from '../../@types/app'
 import { UserSubmission, Challenge } from '../../@types/challenge'
-import GET_APP from '../../graphql/queries/getApp'
+import USER_INFO from '../../graphql/queries/userInfo'
 import ProfileLessons from '../../components/ProfileLessons'
 import ProfileImageInfo from '../../components/ProfileImageInfo'
 import ProfileSubmissions from '../../components/ProfileSubmissions'
-import withQueryLoader, {
-  QueryDataProps
-} from '../../containers/withQueryLoader'
 
 export type UserInfo = {
   username: string
   firstName: string
   lastName: string
 }
 
-const UserProfile: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
-  const { lessons, session } = queryData
-  const fullname = _.get(session, 'user.name', '')
-  const username = _.get(session, 'user.username', '')
-
+const UserProfile: React.FC = () => {
+  const router = useRouter()
+  const username = router.query.username as string
+  const { loading, error, data } = useQuery(USER_INFO, {
```

Comment:
use the custom `use` query that is generated by code-gen

Reply:
References:
https://github.com/garageScript/c0d3-app/blob/4c3f4c0c6d89010fe708021d24f66b5c507f2df4/components/AppNav.tsx#L4
https://github.com/garageScript/c0d3-app/blob/4c3f4c0c6d89010fe708021d24f66b5c507f2df4/components/AppNav.tsx#L54